interactive_engine/bin/giectl.sh

```diff
@@ -0,0 +1,396 @@
+set -e
+set -o pipefail
+readonly WORKSPACE=${GRAPHSCOPE_HOME}
+
+err() {
+  echo -e "${RED}[$(date +'%Y-%m-%dT%H:%M:%S%z')]: [ERROR]${NC} $*" >&2
+}
+
+warning() {
+  echo -e "${YELLOW}[$(date +'%Y-%m-%dT%H:%M:%S%z')]: [WARNING]${NC} $*" >&1
+}
+
+log() {
+  echo -e "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $*" >&1
+}
+
+
+##########################
+# Output useage information.
+# Globals:
+#   None
+# Arguments:
+#   None
+##########################
+usage() {
+cat <<END
+  A script to install dependencies of GraphScope or deploy GraphScope locally.
+
+  Usage: deploy_local [options] [command]
+
+  Options:
+
+    --prefix <path>      install prefix of GraphScope, default is /usr/local
+    --verbose            ouput the debug logging information
+    -h, --help           output help information
+
+  Commands:
+
+    install_deps         install dependencies of GraphScope
+    deploy               deploy GraphScope locally
+END
+}
+
+
+
+set_common_envs() {
+  object_id=$1
+  if [ ! -n "$GRAPHSCOPE_RUNTIME" ]; then
+    export GRAPHSCOPE_RUNTIME=/tmp/graphscope/runtime
+  fi
+
+  if [ -n "$object_id" ]; then
+    export LOG_DIR=$GRAPHSCOPE_RUNTIME/logs/$object_id
+    export CONFIG_DIR=$GRAPHSCOPE_RUNTIME/config/$object_id
+    export PID_DIR=$GRAPHSCOPE_RUNTIME/pid/$object_id
+  fi
+}
+
+# parallel run for k8s
+parallel_run() {
+  run_cmd=$1
+  pod_hosts=$2
+  engine=$3
+  _id=1
+  for pod in `echo $pod_hosts`
+  do
+    kubectl --namespace=$ENGINE_NAMESPACE exec $pod -c $engine -- sh -c "export server_id=$_id && $run_cmd"
+    let _id+=1
+  done
+}
+
+random_generator() {
+  min=$1
+  range=$2
+  _port=$(( ((RANDOM<<15)|RANDOM) % $range + $min ))
+  echo ${_port}
+}
+
+start_coordinator() {
+  JAVA_OPT="-server
+            -Xmx1024m
+            -Xms1024m
+            -XX:+HeapDumpOnOutOfMemoryError
+            -XX:HeapDumpPath=./java.hprof
+            -verbose:gc
+            -Xloggc:${LOG_DIR}/maxgraph-coordinator.gc.log
+            -XX:+PrintGCDetails
+            -XX:+PrintGCDateStamps
+            -XX:+PrintHeapAtGC
+            -XX:+PrintTenuringDistribution
+            -Djava.awt.headless=true
+            -Dsun.net.client.defaultConnectTimeout=10000
+            -Dsun.net.client.defaultReadTimeout=30000
+            -XX:+DisableExplicitGC
+            -XX:-OmitStackTraceInFastThrow
+            -XX:+UseG1GC
+            -XX:InitiatingHeapOccupancyPercent=75
+            -Dfile.encoding=UTF-8
+            -Dsun.jnu.encoding=UTF-8
+            -Dlogfilename=${LOG_DIR}/maxgraph-coordinator.log
+            -Dlogbasedir=${LOG_DIR}/coordinator
+            -Dlog4j.configurationFile=file:${WORKSPACE}/config/log4j2.xml
+            -classpath ${WORKSPACE}/0.0.1-SNAPSHOT/conf/*:${WORKSPACE}/0.0.1-SNAPSHOT/lib/*:"
+  inner_config=$CONFIG_DIR/coordinator.application.properties
+  cp $WORKSPACE/config/coordinator.application.properties $inner_config
+  sed -i "s/ZOOKEEPER_PORT/$2/g" $inner_config
+  pushd $WORKSPACE/coordinator/target/classes/
```

Comment:
Could you help to double check about this? Since all the java code are packaged as jars, do we need to find a specific java class and run?

Reply:
fixed